### DISCUSSION TITLE
Should vaildate_assignment also call validate classmethods for classes inheriting builtins?

### DISCUSSION THREAD

**Discussion by thraizz** [pydantic/pydantic#3063] [Question] [answered]
I am currently trying to validate a class which has an attribute that inherits the `str` built-in. It has an `@classmethod` decorator set so that the value gets checked. This works as intended on instantiation of an object, but not with setting attributes after the creation. This goes against my unedrstanding of `validate_assignment`, but maybe I am just using it wrong.
Here's an example:
```python
from pydantic import BaseModel, ValidationError, validator

class Bool(str):
    @classmethod
    def validate(cls, value):
        if isinstance(value, bool):
            return value
        if isinstance(value, str):  # or isinstance(value, unicode):
            if value == 'true':
                return True
            if value == 'false':
                return False
        raise ValidationError("Not a valid bool!")

class Test(BaseModel):
    class Config:
        validate_assignment = True
    valid: Bool

b = Test(valid="true")
# This should raise an ValidationError in my understanding, but it doesn't.
b.valid = "somethingelse"

```

Am I just doing something wrong here? I know that this is not the main intend for pydantic, but its just an requirement for our project and much more fitting than our current custom-written solution.

**Comment by PrettyWood** [comment, marked as answer]
Hi @thraizz 
Your `Bool` class never triggers validation. You can write `Test(valid="pika")` and it will still work.
Why? Because you inherit from `str` which is already validated. If you want to overwrite it it's a bit more complicated.
But in your case just write
```py
class Test(BaseModel):
    class Config:
        validate_assignment = True

    valid: bool
```
Pydantic will understand `"true"` as `True` ;)
Hope it helps

**Reply by thraizz** [reply]
Ah, I see, this explains the behavior I'm seeing. The other important thing I was missing is that the class which should check the builtin needs 
```python3
    @classmethod
    def __get_validators__(cls):
        yield cls.validate
```
otherwise the assignments wont be validated as expected.
Thanks for the quick reply!
